### PR TITLE
feat: implement `dss stop` command

### LINKED ISSUE #65: Implement DSS `stop` command
### Why it needs to get done

The stop command stops a specified notebook within the DSS environment. This command is expect to start a stopped notebook

```
dss stop my-notebook
```

Please use the UX [spec](https://docs.google.com/document/d/1v0wU8TBb-ui6RcjGRbLwA6XFZnWbuiVb2TPmEgwKejY/edit#heading=h.rfcm5y81sk1z) documentation .

### What needs to get done

1. Implement stop command
2. Implement integration test with Microk8s

### When is the task considered done

1. Command is implemented
2. Tests are passing

### PR DESCRIPTION
**Pull request by NohaIhab**
closes #65

### CODE REVIEW

File: tests/integration/test_dss.py [hunk 1]
```diff
@@ -130,6 +130,37 @@ def test_log_command(cleanup_after_initialize) -> None:
     assert "Starting gunicorn" in result.stderr
 
 
+def test_stop_notebook(cleanup_after_initialize) -> None:
+    """
+    Tests that `dss stop` successfully stops a notebook as expected.
+
+    Must be run after `dss create`.
+    """
+    kubeconfig_file = "~/.kube/config"
```
Comment (DnPlas) [inline tests/integration/test_dss.py:139]: I think at this point we can take out this and create a global variable. I see other test cases are using it.

File: src/dss/stop.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        name (str): The name of the notebook server.
+        lightkube_client (Client): The Kubernetes client.
+    """
+
+    if not does_notebook_exist(
+        name=name, namespace=DSS_NAMESPACE, lightkube_client=lightkube_client
+    ):
+        logger.warn(f"Failed to stop Notebook. Notebook {name} does not exist.")
+        logger.info("Run 'dss list' to check all notebooks.")
+        exit(1)
```
Comment (misohu) [inline src/dss/stop.py:29]: Based on our last daily sync we agreed to put the exit code handling into main.

File: src/dss/stop.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    Args:
+        name (str): The name of the notebook server.
+        lightkube_client (Client): The Kubernetes client.
+    """
+
+    if not does_notebook_exist(
+        name=name, namespace=DSS_NAMESPACE, lightkube_client=lightkube_client
+    ):
+        logger.warn(f"Failed to stop Notebook. Notebook {name} does not exist.")
```
Comment (misohu) [inline src/dss/stop.py:27]: This should be error ?

**Review by misohu — COMMENTED** [review]
Small comments Good job.

**Pull request merged**